### ISSUE TITLE
IDE0004 False positive

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#33043]
**Version Used**: VS 2019 preview 2

**Steps to Reproduce**:

```csharp
if ((int?)1 is null)
{
}
```

**Expected Behavior**:

No diagnostic is reported

**Actual Behavior**:

IDE0004 is reported. If you apply the fix (remove the cast), the code doesn't compile.
